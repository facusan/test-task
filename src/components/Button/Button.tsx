import styled from "styled-components";

export type ButtonProps = {
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  children?: React.ReactNode;
  color?: "primary" | "secondary";
  disabled?: boolean;
  size?: "small" | "large";
};

export const Button = styled.button<ButtonProps>`
  background: ${(props) => (props.color === "primary" ? "#167EFF" : "white")};
  color: ${(props) => (props.color === "primary" ? "white" : "black")};
  border-color: ${(props) => (props.color === "primary" ? "#167EFF" : "black")};
  width: ${(props) => (props.size === "small" ? "200px" : "300px")};
  padding: 20px;
  border-radius: 6px;
  margin: 10px;
  font-weight: bold;
  font-family: sans-serif;
  font-size: large;
`;
