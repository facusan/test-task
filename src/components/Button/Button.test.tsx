import renderer from "react-test-renderer";
import { Button } from "./Button";

it("renders primary Button correctly", () => {
  const component = renderer.create(
    <Button onClick={() => {}} color={"primary"}>
      Primary
    </Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders secondary Button correctly", () => {
  const component = renderer.create(
    <Button onClick={() => {}} color={"secondary"}>
      Secondary
    </Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders small Button correctly", () => {
  const component = renderer.create(
    <Button onClick={() => {}} color={"secondary"} size="small">
      Small
    </Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders large Button correctly", () => {
  const component = renderer.create(
    <Button onClick={() => {}} color={"secondary"} size="large">
      Small
    </Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
