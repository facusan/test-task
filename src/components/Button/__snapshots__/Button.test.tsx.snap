// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders large Button correctly 1`] = `
<button
  className="sc-bczRLJ jclefq"
  color="secondary"
  onClick={[Function]}
  size="large"
>
  Small
</button>
`;

exports[`renders primary Button correctly 1`] = `
<button
  className="sc-bczRLJ lniJCu"
  color="primary"
  onClick={[Function]}
>
  Primary
</button>
`;

exports[`renders secondary Button correctly 1`] = `
<button
  className="sc-bczRLJ jclefq"
  color="secondary"
  onClick={[Function]}
>
  Secondary
</button>
`;

exports[`renders small Button correctly 1`] = `
<button
  className="sc-bczRLJ lhoYHb"
  color="secondary"
  onClick={[Function]}
  size="small"
>
  Small
</button>
`;
